serrano/resources/stats.py

```diff
@@ -81,27 +111,36 @@ def get(self, request):
             if verbose_name_plural.islower():
                 verbose_name_plural = verbose_name_plural.title()
 
-            # Get count from cache or database
-            label = ':'.join([opts.app_label, opts.module_name, 'count'])
-            key = cache_key(label, kwargs={'queryset': queryset})
-
-            if params['refresh']:
-                count = None
-            else:
-                count = cache.get(key)
-
-            if count is None:
-                count = queryset.values('pk').distinct().count()
-                cache.set(key, count, timeout=NEVER_EXPIRE)
-
+            # Placeholder with the model name. The count will be replaced if
+            # successful.
             data.append({
-                'count': count,
+                'count': None,
                 'app_name': app_name,
                 'model_name': model_name,
                 'verbose_name': verbose_name,
                 'verbose_name_plural': verbose_name_plural,
             })
 
+            # Asynchronously execute the count
+            result = pool.apply_async(get_count, args=(
+                request,
+                model,
+                params['refresh'],
+                QueryProcessor,
+                context
+            ))
+
+            results.append(result)
+
+        pool.close()
+
+        for i, r in enumerate(results):
+            try:
+                count = r.get(timeout=settings.STATS_COUNT_TIMEOUT)
```

Comment:
The one thing to consider with this timeout is that it won't obviously cancel the actual query running in the database, it just allows the response to return.